src/components/ToggleElementDropdown/ToggleElementDropdown.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import core from 'core';
+
+import selectors from 'selectors';
+import actions from 'actions';
+
+import { zoomTo } from 'helpers/zoom';
+
+import ToggleElementButton from '../ToggleElementButton';
+
+import './ToggleElementDropdown.scss';
+
+class ToggleElementDropdown extends React.PureComponent {
+
+  constructor(props){
+    super(props);
+    this.state = { zoom: Math.ceil(core.getZoom() * 100), value: 100 };
+    this.textarea = React.createRef();
+  }
+  componentDidMount(){
+    core.addEventListener('zoomUpdated', this.onZoomUpdated);
```

Comment:
We are already listening this event in eventHandler.js, line 35, `core.addEventListener('zoomUpdated', onZoomUpdated);`. So every time this event is triggered, an action will be dispatched(see `onZoomUpdated.js`) to update the store. What we can do here is connect this component with redux and get the zoom value from `state.viewer.zoom`. Let me know if this is confusing.